platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -341,6 +355,71 @@ MGL_EXPORT
  */
 @property (nonatomic, readonly) unsigned long long countOfBytesCompleted;
 
+
+#pragma mark - Managing Ambient Cache
+
+/**
+ Sets the maximum ambient cache size in megabytes. The default maximum cache
+ size is 50 MB. To disable ambient caching, set the maximum ambient cache size
+ to `0`. Setting the maximum ambient cache size does not impact the maximum size
+ of offline packs.
+ 
+ While this method does not limit the space available to offline packs,
+ data in offline packs count towards this limit. If the maximum ambient
+ cache size is set to 30 MB and 20 MB of offline packs are downloaded,
+ there may be only 10 MB reserved for the ambient cache.
+ 
+ This method should be called before the map and map style have been loaded.
+ 
+ This method is potentially expensive, as the database will trim cached data
+ in order to prevent the ambient cache from being larger than the
+ specified amount.
+ 
+ @param cacheSize The maximum size in bytes for the ambient cache.
+ @param completion The completion handler to call once the maximum ambient cache size
+ has been set. This handler is executed synchronously on the main queue.
+ */
+
+- (void)setMaximumAmbientCacheSize:(NSUInteger)cacheSize withCompletionHandler:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ Invalidates the ambient cache. This method checks that the tiles in the
+ ambient cache match those from the server. Local tiles that do not match
+ the latest version on the server are updated.
+ 
+ This is more efficient than cleaning the cache because valid local tiles will
+ not be downloaded again.
+ 
+ Resources shared with offline packs will not be affected by this method.
+ 
+ @param completion The completion handler to call once the ambient cache has
+ been revalidated. This handler is executed asynchronously on the main queue.
+ */
+
+- (void)invalidateAmbientCacheWithCompletionHandler:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ Clears the ambient cache, freeing storage space. This method does not
```

Comment:
Change to "Deletes the ambient cache."

Let's leave out "freeing storage space" -- developers will be more incentivized to use this method than they should be.

Reply:
Should we also mention the reinitialization of the ambient cache here?

Reply:
Per an earlier review from @friedbunny, I would like to keep it as `clears` so that we are starting the documentation with the verb from method name. How about `Clears the ambient cache by deleting resources.`?

Reply:
Ah okay, fair enough. The core documentation consistently uses "erase" as the verb but the method name does start with `clear`, and I understand the desire to keep those consistent. I'm also ok with using "erase" as a verb instead of "delete" to match the core documentation:

> Clears the ambient cache by erasing resources from the ambient cache.

FWIW I think the rest of the core documentation is good to copy and paste over to the platform side:

> This operation can be potentially slow because it will trigger a VACUUM on SQLite, forcing the database to move pages on the filesystem.

> Resources overlapping with offline regions will not be affected by this call.

Reply:
Core and Android use `clearAmbientCache` (unless I'm looking at the wrong thing).